pyproject.toml

```diff
@@ -0,0 +1,21 @@
+[project]
+name = "htm.core"
+version = "2.1.16"
+description = "This is a Community Fork of the nupic.core"
+readme = "README.md"
+requires-python = ">=3"
+dependencies = [
+    "numpy",
+    "hexy",
+    "pybind11",
+    "pytest-cov",
+    "hexy",
+    "mock",
+    "cmake",
+    "requests",
+    "prettytable",
+    "setuptools>=75.8.0",
+]
+
+[build-system]
```

Comment:
I thought that [build-system] required a "build-backend" but maybe the default is setuptools.

Reply:
Because uv seems to run in a temporary virtual environment, the errors gave me clue what was missing from the pyproject,toml to populate that environment. Once that worked I went back to a conda environment to check if the same fix worked there.
As I mentioned earlier, I managed with the minimum of changes. I
 just hope that it works as well in other platforms.

Reply:
Since I haven't had a chance to try out your changes I can't really comment, but off the top of my head I would think that using setuptools and pip would be the more Pythonic way to go than with scikit-build-core.

Reply:
However, because you indicated that you cleaned up some messy hacks in the C++ build code, maybe you can incorporate those changes with mine within the setuptools framework.